terminfo/dynamic/dynamic.go

```diff
@@ -117,7 +117,7 @@ func unescape(s string) string {
 }
 
 func (tc *termcap) setupterm(name string) error {
-	cmd := exec.Command("infocmp", "-1", name)
+	cmd := exec.Command("infocmp", "-x", "-1", name)
```

Comment:
So building this winds up depending on non-standardized terminfo names.  While I can appreciate the desire to do this, it's not something that I'd prefer to do.  The dynamic mode is meant to be a fall back / failsafe, and I worry that using this approach may de-value that.

Having said all that, I *am* amenable to having the clipboard string added, and I'd really like it to be added to the terminals that can do this, rather than just doing the dynamic thing here.

I'm going to look at alternative approaches, which might mean creating a new variable or flag to enable most of the modern xterm-isms.

Reply:
Honestly I think the issue is more the built-in shipped copies of escape codes. That is exactly what terminfo is for (which is why I changed my local tcell copy to always default to dynamically requesting them). If the system doesn't provide proper terminfo descriptions for the used terminal (or the user prefers lying about their TERM var) the system should be fixed. IMHO having these kind of hardcoded manual terminfo tables around in applications just unnecessarily complicates things.

Reply:
XTGETTCAP looks cool.  I will look into it.  Unfortunately with a lot of these things we wind up having a very mixed set of things... and of course the problem is that terminal capability names are limited because a lot of features don't have standard names yet.

Reply:
I looked at XTGETTCAP a bit and tried in a variety of different terminal emulators.  The results weren't very encouraging.

In a perfect world terminfo would indeed be able to provide all this information.  But the problem with terminfo is that it's kind of stuck in some 1990's mode of thinking, and new capabilities rarely are added, and sometimes are added using incompatible sequences, which breaks the whole point of having a single database to look at.

The other thing is that you might find yourself using an editor that lacks a terminfo entry for the emulator (or even real terminal) that you're using.

The upshot of all that is that we cannot rely on terminfo for this.  What's interesting is that most of the newer editors that are actually implementing things are doing so collaboratively, so that escape sequences are mostly added using OSC sequences that don't conflict with one another, and will be ignored by systems that don't understand them.  It's not *perfect*, and there are some conflicts, but it's definitely better than if they weren't talking at all to each other.